[CI] Created new CI job pipeline for Navi4x architecture

stefankoncarevic opened the pull request:
This PR adds a new CI pipeline specifically for the Navi4x architecture. Most configurations and functionalities have been adapted from the main Jenkins file, with adjustments for Navi4x. Created stages for setting system properties, environment setup, and executing PR CI tests and E2E tests. 
Please review the Jenkins file and provide any suggestions for further improving the testing process.

stefankoncarevic committed:
Create new CI pipeline for Navi4x architecture

Implemented a dedicated Jenkins pipeline for Navi4x architecture.
Added stages for setting system properties, environment setup,
and running PR CI tests and E2E tests.

manupak commented:
Thanks @stefankoncarevic . To me, this feels much cleaner.
Would you be able to set up a cron job using this branch + jenkinsfile ?

stefankoncarevic commented:
That's right, I take this branch and the script I made and run CI, so I debug if there are any errors.

stefankoncarevic committed:
Fix: Handle unexpected output from rocm_agent_enumerator -name
on navi4x-hw-31 CI machine.

The rocm_agent_enumerator -name command on the navi4x-hw-31 CI
machine does not return the expected list of agents. Instead,
it returns a timeout message that does not contain any
"gfx" values. This commit modifies the get_agents function to
handle this case by falling back to using rocm_agent_enumerator
without the -name option if the output does not contain
any "gfx" values.

- Added a check to ensure the output contains "gfx" values.
- Fallback to rocm_agent_enumerator without -name if
  the output is invalid.
- Filtered out "gfx000" from the list of agents.

This change ensures that the correct agents are identified and
used in the testing process.

krzysz00 commented on mlir/utils/jenkins/Jenkinsfile.navi4x:
```diff
@@ -0,0 +1,163 @@
+}
+
+void setHeartbeat() {
+    script {
```
Why all these `script {}` blocks? I'm pretty sure we don't need them?

stefankoncarevic replied:
Not sure. Copied from our nightly. I assumed it was needed or at least wouldn't hurt. I will remove it.

stefankoncarevic committed:
Removed unnecessary script {} blocks when setting the heartbeat property within the parameter sweep job because we don't running that stage on navi4x.

pcf000 commented:
Looks good to me.  I know there's a way to share common code among Jenkinsfiles;  with three now, we should use it, but that's outside the scope here.
